src/bindings/html/langs.js

```diff
@@ -18,7 +18,7 @@ export function getAdditionalLanguages() {
 export function onlanguagechage(
   appVersion, defaultLang, availableLangs, requestedLangs) {
 
-  return this.languages = Promise.all([
+  return Promise.all([
```

Comment:
ok, I fixed that (together with passing langs to initViews), but there's a bug in result, that I'd like to learn how to tackle in this code.

Basically, if we are in a pretranslated scenario, then we should not need `langs` in `changeLanguage`. We should be able to use prevLangs vs. newLangs to establish that we don't need changes, call `initViews` with pretranslated=true and do nothing else.

But because of the chain assignments in langs.on* and in `init`, we have to return langs, and we need to assign them to `this.languages`. How can we refactor that code to allow for the fastpath?